cmd/lingo/main.go

```diff
@@ -182,7 +182,11 @@ func run() error {
 	}()
 	go func() {
 		setupLog.Info("Starting leader election")
-		le.Start(ctx)
+		err := le.Start(ctx)
+		if err != nil {
+			setupLog.Error(err, "starting leader election")
+			os.Exit(1)
```

Comment:
When the context is canceled on a graceful shutdown (sigterm) then `Start()` would return the "cancel" error that results in an `Exit(1)`. 
This error handling strategy is used in other places in this file as well. Instead, we could wrap the context in L110 
`ctx, cancel := context.WithCancel(ctrl.SetupSignalHandler())` and call `cancel()` instead. WDYT?

Reply:
I didn't fully understand how to change it. Could you elaborate? Are you saying instead of calling os.Exit() just call cancel()?

I didn't see any other example of using cancel on the context in main.go. I tried to follow the same pattern as the other start functions.

Reply:
The only error returned from `Start()` is coming from the context, now. Thanks to your for loop. Calling cancel on the context again is not needed. I would not return an error in `Start()` and let the Go routine complete. The main thread should handle the shutdown. I should have deleted my first comment. Sorry for the confusion. 
To elaborate for a different use case where other errors can happen, this is an example:
```go
	ctx, cancel := context.WithCancel(ctrl.SetupSignalHandler())
	go func() {
		// in subroutine
		if err:= doSomething(ctx); err!=nil{
			// todo: log error
			cancel()
		}
	}()
	// main thread
	if err:=doSomethingElse(ctx);err!=nil{
		// todo: log error
		os.Exit(1)
	}
```
Please note that the error is not returned to the main thread.

Reply:
Thanks for that example! I think I get it now. To me it feels cleaner to just directly exit instead of cancelling a context which then may in the future cause an exit? I favor being explicit and going to the exit path as soon as it makes sense.